reana_db/models.py

```diff
@@ -812,6 +833,73 @@ def __repr__(self):
         return "<AuditLog {} {}>".format(self.id_, self.action)
 
 
+class WorkspaceRetentionRuleStatus(enum.Enum):
+    """Enumeration of workspace retention rule status."""
+
+    created = 0
+    active = 1
+    inactive = 2
+    applied = 3
+
+
+class WorkspaceRetentionAuditLog(Base):
+    """Workspace Retention audit log table."""
+
+    __tablename__ = "workspace_retention_audit_log"
+    __table_args__ = {"schema": "__reana"}
+
+    workspace_retention_rule_id = Column(
+        UUIDType,
+        ForeignKey("__reana.workspace_retention.id_"),
+        primary_key=True,
+        nullable=False,
+    )
+    timestamp = Column(DateTime, server_default=func.now(), primary_key=True)
+    action = Column(
+        Enum(WorkspaceRetentionRuleStatus), primary_key=True, nullable=False
+    )
+
+    def __repr__(self):
+        """Workspace Retention audit log string representation."""
+        return f"<WorkspaceRetentionAuditLog {self.workspace_retention_rule_id} {self.action}>"
+
+
+class WorkspaceRetention(Base):
```

Comment:
We have three new classes with this PR:

- class WorkspaceRetentionRuleStatus(enum.Enum):
- class WorkspaceRetentionAuditLog(Base):
- class WorkspaceRetention(Base):

Quick impressions seeing these names: the first is nicely self-understandable, the third seem to be unfinished (retention what?) as if calling for a final noun (WorkspaceRetentionRule?), and the second is perhaps too businessy if we include "audit" (WorkspaceRetentionLogs?) and may perhaps clash with the AuditLog table that we have already for some user actions. 

(BTW it may be a good idea to reuse the existing AuditLog one, so that we have centralised all the "auditable actions" in one place... but that would need more changes.)